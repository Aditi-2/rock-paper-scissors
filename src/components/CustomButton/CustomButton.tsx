import styled from "@emotion/styled";
import React from "react";

interface CustomButtonProps {
  label: string;
  onClick?: () => void | Promise<void>;
}
export const CustomButton: React.FC<CustomButtonProps> = ({
  onClick,
  label,
}) => {
  return (
    <CustomButtonStyled className="custom-btn ripple" onClick={onClick}>
      {label}
    </CustomButtonStyled>
  );
};

const CustomButtonStyled = styled.button`
  width: 100%;
  padding: 4px;
  margin: 5px 0;
  font-size: 0.5rem;
  max-width: 9rem;
  cursor: pointer;
  box-shadow: 0 0 4px #999;
  outline: none;
  border: none;
  background-position: center;
  transition: background 0.8s;

  @media only screen and (min-width: 800px) {
    padding: 1rem;
    font-size: 1.2rem;
    max-width: 9rem;
  }
  &.ripple:hover {
    background: #dadada radial-gradient(circle, transparent 1%, #dadada 1%)
      center/15000%;
  }
  &.ripple:active {
    background-color: #222222;
    background-size: 100%;
    transition: background 0s;
  }
`;
