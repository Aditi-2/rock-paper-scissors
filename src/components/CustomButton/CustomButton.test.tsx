import React from "react";
import { create } from "react-test-renderer";
import { fireEvent, render, screen } from "@testing-library/react";
import { CustomButton } from "./CustomButton";

const onClick = jest.fn();

const getComponent = () => <CustomButton label="button" onClick={onClick} />;

test("CustomButton Component snapshot", () => {
  const component = create(getComponent()).toJSON();
  expect(component).toMatchSnapshot();
});

test("renders CustomButton Component", () => {
  render(getComponent());
  const button = screen.getByText(/button/i);
  expect(button).toBeInTheDocument();
});

test("CustomButton Component onClick is called", () => {
  render(getComponent());
  const button = screen.getByText(/button/i);
  fireEvent.click(button);
  expect(onClick).toHaveBeenCalled();
});
