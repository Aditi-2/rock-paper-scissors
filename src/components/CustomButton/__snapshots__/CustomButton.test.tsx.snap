// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomButton Component snapshot 1`] = `
<button
  className="custom-btn ripple css-zsodnc"
  onClick={[MockFunction]}
>
  button
</button>
`;
